asmcomp/cmmgen.ml

```diff
@@ -1376,6 +1376,14 @@ let check_bound unsafe dbg a1 a2 k =
 let default_prim name =
   Primitive.simple ~name ~arity:0(*ignored*) ~alloc:true
 
+let int64_native_prim name arity =
+  let u64 = Unboxed_integer Pint64 in
+  let rec make_args = function 0 -> [] | n -> u64 :: make_args (n - 1) in
+  Primitive.make ~name ~native_name:(name ^ "_native")
+    ~alloc:false
```

Comment:
`alloc:false` is incompatible with primitives that can raise (div and mod).  It could be useful to deal with the zero-dividend when generating the cmm code (as in safe_divmod_bi) and use a C-primitives that doesn't raise (and thus can indeed be marked as noalloc), but I'm not sure it's worth the trouble.

Reply:
Oops, indeed. Fixed.